### PR TITLE
Remove temporary build_armcc workaround 

### PR DESCRIPTION
**Pull request by bensze01**
Overriding build_armcc unconditionally would fail if build_armcc didn't exist on the branc being tested.

- Internal CI
  - development, good
      - [PR-head][1] (failures due to unrelated CI instability)
      - [PR-merge][2]
      - [release][15]
  - mbedtls-3.6
      - [PR-head][3]
      - [PR-merge][4]
      - [release][16]
  - mbedtls-2.28, good
      - [PR-head][5] (failure due to unrelated CI instability)
      - [PR-merge][6]
      - [release][17]
  - [CMake Fail][7]
- OpenCI
  - development, good
      - [PR-head][8]
      - [PR-merge][9]
      - [release][18]
  - mbedtls-3.6
      - [PR-head][10]
      - [PR-merge][11]
      - [release][19]
  - mbedtls-2.28, good
      - [PR-head][12]
      - [PR-merge][13]
      - [release][20]
  - [CMake Fail][14]

[1]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-906-head/94/
[2]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-906-merge/138/
[3]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-1233-head/1/
[4]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-1233-merge/1/
[5]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-850-head/3/
[6]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-850-merge/2/
[7]: https://jenkins-mbedtls.oss.arm.com/job/mbed-tls-pr-ci-testing/view/change-requests/job/PR-759-head/3/
[8]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-906-head/17/
[9]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-906-merge/16/
[10]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-1233-head/1/
[11]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-1233-merge/1/
[12]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-850-head/
[13]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-850-merge/4/
[14]: https://mbedtls.trustedfirmware.org/job/mbed-tls-restricted-pr-ci-testing/view/change-requests/job/PR-759-head/10/
[15]: https://jenkins-mbedtls.oss.arm.com/job/mbedtls-release-ci-testing/598/
[16]: https://jenkins-mbedtls.oss.arm.com/job/mbedtls-release-ci-testing/599/
[17]: https://jenkins-mbedtls.oss.arm.com/job/mbedtls-release-ci-testing/600/
[18]: https://mbedtls.trustedfirmware.org/job/mbedtls-release-ci-testing/203/
[19]: https://mbedtls.trustedfirmware.org/job/mbedtls-release-ci-testing/204/
[20]: https://mbedtls.trustedfirmware.org/job/mbedtls-release-ci-testing/205/

### CODE REVIEW

File: vars/common.groovy [hunk 1]
```diff
@@ -284,8 +284,10 @@ BranchInfo get_branch_information() {
          * and we no longer care about older pull requests, choose
```
Comment (gilles-peskine-arm) [inline vars/common.groovy:284]: This is a temporary workaround, not needed for branches forked since 14 March 2013. So how about we remove the workaround?

**Review by gilles-peskine-arm — APPROVED** [review]
LGTM on code inspection (we need to confirm with CI jobs). This is a workaround for branches forked before https://github.com/Mbed-TLS/mbedtls/pull/7163 (+ backport) was merged, and it was merged 13 months ago.

**Comment by bensze01** [comment]
All the relevant tests have passed - all the failures are caused by an unrelated issue affecting the stability of the internal CI.
@gilles-peskine-arm would you be OK with making this a single-reviewer PR considering its small size?

**Review by mpg — APPROVED** [review]
LGTM

**Pull request merged**